cmd/goal/clerk.go

```diff
@@ -787,9 +787,16 @@ var signCmd = &cobra.Command{
 
 		for _, group := range groupsOrder {
 			txnGroup := []transactions.SignedTxn{}
-			for _, txn := range txnGroups[group] {
+			for idx, txn := range txnGroups[group] {
 				if lsig.Logic != nil {
 					txn.Lsig = lsig
+					if signerAddress != "" {
+						addr, err := basics.UnmarshalChecksumAddress(signerAddress)
+						if err != nil {
+							reportErrorf("Signer invalid txn[%d]: %v", idx, err)
+						}
+						txn.AuthAddr = addr
+					}
```

Comment:
I agree with the logic here, but the `basics.UnmarshalChecksumAddress` operation needs to be done in 
```golang
if lsig.Logic == nil {
			// sign the usual way
			dataDir := ensureSingleDataDir()
			client = ensureKmdClient(dataDir)
			wh, pw = ensureWalletHandleMaybePassword(dataDir, walletName, true)
		} else {
                    // << -- here
                }
```

there is never a good reason to provide invalid signer address, so we should validate that upfront.

Reply:
Makes sense, made the change